Phoenix/PHSpace.m

```diff
@@ -223,8 +238,34 @@ - (void)removeWindows:(NSArray<PHWindow *> *)windows {
 }
 
 - (void)moveWindows:(NSArray<PHWindow *> *)windows {
-    CGSMoveWindowsToManagedSpace(
-        CGSMainConnectionID(), (__bridge CFArrayRef)[self identifiersForWindows:windows], self.identifier);
+    if (![NSProcessInfo isOperatingSystemAtLeastSonoma145]) {
+        CGSMoveWindowsToManagedSpace(
+            CGSMainConnectionID(), (__bridge CFArrayRef)[self identifiersForWindows:windows], self.identifier);
+        return;
+    }
+
+    // CGSMoveWindowsToManagedSpace is broken in MacOS 14.5, so use the legacy Compat ID API instead.
+    [self moveWindowsWithCompatId:windows];
+}
+
+/**
+ * - https://github.com/kasper/phoenix/issues/348
+ * - https://github.com/koekeishiya/yabai/issues/2240#issuecomment-2116326165
+ * - https://github.com/ianyh/Amethyst/issues/1643#issuecomment-2132519682
+ */
+- (void)moveWindowsWithCompatId:(NSArray<PHWindow *> *)windows {
```

Comment:
Can you move this method before `moveWindows`? I prefer having functions being defined in logical order.

Reply:
Done!